### PR TITLE
Support CREATE VIEW and CREATE TABLE verification

### PR DESCRIPTION
**Pull request by caithagoras**
## Verify CREATE VIEW
If the specified view already exists, create a temporary view to match
the existing view. Otherwise, do nothing for setup queries. Rewrite the
target view name, and run both control and test queries. Run a SHOW
CREATE VIEW query and the returned CREATE VIEW statement needs to match.

## Verify CREATE TABLE
Rewrite the target table of the CREATE TABLE statement, run both
control and test queries, run SHOW CREATE TABLE query as the check.

```
== RELEASE NOTES ==

Verifier Changes
* Add support to verify ``CREATE VIEW`` and ``CREATE TABLE`` queries.
```

### CODE REVIEW

**Comment by sujay-jain** [comment]
started reviewing - first 3 commits LGTM

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DdlVerification.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    protected DeterminismAnalysisDetails analyzeDeterminism(QueryObjectBundle control, DdlMatchResult matchResult)
+    {
+        throw new UnsupportedOperationException("analyzeDeterminism is not supported for DdlVerification");
+    }
+
+    @Override
+    protected Optional<String> resolveFailure(Optional<QueryObjectBundle> control, Optional<QueryObjectBundle> test, QueryContext controlQueryContext, Optional<DdlMatchResult> matchResult, Optional<Throwable> throwable)
+    {
+        return Optional.empty();
+    }
```
Comment (sujay-jain) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DdlVerification.java:119]: Do we not want a ```failureResolveManager``` like DataVerification?

File: presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java [hunk 2]
```diff
@@ -166,11 +173,43 @@ public QueryObjectBundle rewriteQuery(@Language("SQL") String query, ClusterType
...
+            CreateView createView = (CreateView) statement;
+            QualifiedName temporaryViewName = generateTemporaryName(Optional.empty(), prefix);
+            ImmutableList.Builder<Statement> setupQueries = ImmutableList.builder();
+
+            // Check to see if there is an existing view with the specified view name.
+            // If view exists, create a temporary view that are has the same definition as the existing view.
+            // Otherwise, do not pre-create temporary view.
+            try {
+                CreateView createExistingView = (CreateView) sqlParser.createStatement(
+                        getOnlyElement(prestoAction.execute(new ShowCreate(VIEW, createView.getName()), REWRITE, SHOW_CREATE_VIEW_CONVERTER).getResults()),
```
Comment (sujay-jain) [inline presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java:186]: Should we extract this into a method?

**Review by sujay-jain — APPROVED** [review]
LGTM

File: presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DdlVerification.java (hunk 1 above)
Comment (caithagoras) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/framework/DdlVerification.java:119]: Failure resolver is used to automatically resolve verification failures due to (test / checksum) query failures. For ddl verification, we can start with no auto resolution, and see if we actually see any query failures are actually false alerts.

File: presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java (hunk 2 above)
Comment (caithagoras) [reply presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/QueryRewriter.java]: This method would only be used once if we extract. I'm changing this to 2 statements - that might help readability.

**Comment by caithagoras** [comment]
@sujay-jain Thanks for reviewing!

**Pull request merged**